@here/harp-mapview/lib/MapView.ts

```diff
@@ -1452,10 +1452,14 @@ export class MapView extends THREE.EventDispatcher {
      */
     get pixelToWorld(): number {
         if (this.m_pixelToWorld === undefined) {
-            const { width, height } = this.getCanvasClientSize();
-            const center = this.getWorldPositionAt(width * 0.5, height);
-            const unitOffset = this.getWorldPositionAt(width * 0.5 + 1.0, height);
-
+            const size = this.getCanvasClientSize();
+            const { centerX, centerY } = { centerX: size.width * 0.5, centerY: size.height * 0.5 };
+            // Calculate canvas middle point (center of projection) world position.
+            const center = this.getWorldPositionAt(centerX, centerY);
+            // Calculate world point that is offset by one pixel in x-axis (or camera right vector)
+            // direction, this ensures that we are perspective/tilt indepenent.
+            const unitOffset = this.getWorldPositionAt(centerX + 1.0, centerY);
+            // Our ratio equals the distance between points in the world space.
             this.m_pixelToWorld =
```

Comment:
Couldn't we derive the ratio from the zoomlevel instead of doing raycasting at the screen center?
It might happen that we don't hit the earth surface especially if you tilt using middle mouse button and also later when we use the SphereProjection.

Reply:
This approach is tilt independent, as also will work with spherical projection. Of course we have the problem if tilt constraints would be removed (possibility to hit outer space instead of terrain), for that case we have discussed to use far plane as the ray cast limit, but there is another problem is if roll would be possible and X screen space offset would point the ray far on the (rotated) horizon.